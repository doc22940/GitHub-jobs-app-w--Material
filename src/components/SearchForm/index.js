import React from "react";
import Button from "@material-ui/core/Button";
import Input from "@material-ui/core/Input";
import Grid from "@material-ui/core/Grid";

const SearchForm = ({ bindDescription, bindLocation, handleSubmit }) => {
  return (
    <form onSubmit={handleSubmit}>
      <Grid container direction="row" spacing={3}>
        <Grid item xs={12} sm={5}>
          <Input {...bindDescription} fullWidth />
        </Grid>

        <Grid item xs={12} sm={5}>
          <Input {...bindLocation} fullWidth />
        </Grid>
        <Grid item xs={12} sm={2}>
          <Button
            variant="contained"
            type="submit"
            fullWidth
            color="primary"
            size="small"
          >
            Search
          </Button>
        </Grid>
      </Grid>
    </form>
  );
};

export default SearchForm;
