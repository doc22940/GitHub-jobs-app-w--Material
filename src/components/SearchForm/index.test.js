import React from "react";
import renderer from "react-test-renderer";

import SearchForm from "./";

describe("SearchForm", () => {
  test("snapshot renders", () => {
    const handleSubmit = jest.fn();
    const component = renderer.create(
      <SearchForm handleSubmit={handleSubmit} />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
