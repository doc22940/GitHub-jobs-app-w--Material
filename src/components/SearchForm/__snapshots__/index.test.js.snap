// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm snapshot renders 1`] = `
<form
  className="makeStyles-root-1"
>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-5"
    >
      <div
        className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth"
        onClick={[Function]}
        state=""
      >
        <input
          className="MuiInputBase-input MuiInput-input"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Description"
          type="text"
        />
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-5"
    >
      <div
        className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth"
        onClick={[Function]}
        state=""
      >
        <input
          className="MuiInputBase-input MuiInput-input"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Location"
          type="text"
        />
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-2"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-containedSizeSmall MuiButton-sizeSmall MuiButton-fullWidth"
        disabled={false}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="submit"
      >
        <span
          className="MuiButton-label"
        >
          Search
        </span>
      </button>
    </div>
  </div>
</form>
`;
